Tasks/Lobach/task3.py

```diff
@@ -0,0 +1,127 @@
+            if h == 0 or h == 12:
+                dh[h + 1] = 'час'
+            if 60 - m == 1:
+                space = space.replace('а', 'ой')
+                print(f'без {space} минуты {dh[h + 1]}')  
+            if 60 - m == 2:
+                space = space.replace('е', 'ух')
+                print(f'без {space} минут {dh[h + 1]}')
+            if 60 - m == 3:
+                space = space.replace('и', 'ех')
+                print(f'без {space} минут {dh[h + 1]}')
+            if 60 - m == 4:
+                space = space + 'х'
+                print(f'без {space} минут {dh[h + 1]}')
+            if 60 - m == 8:
+                space = space.replace('емь', 'ьми')
+                print(f'без {space} минут {dh[h + 1]}')
+            if 5 <= 60 - m <= 7 or 9 <= 60 - m <=15:
+                space = space.replace('ть', 'ти')
+                print(f'без {space} минут {dh[h + 1]}')
+elif a == 'N':
+    x = datetime.now()
+    firstdigit = x.minute // 10
+    lastdigit = x.minute % 10
+    space = ''
+    
+    if lastdigit  == 1:
+        space = 'минута'
+    if 1 < lastdigit <= 4:
+        space = 'минуты'
+    if lastdigit > 4 or (lastdigit  == 1 and firstdigit == 1):
+        space = 'минут'
+    
+    if x.minute == 0:
+        if 4 < x.hour < 13 or 16 < x.hour <= 24:
+                space = 'часов'
+        if 1 < x.hour <= 4 or 13 < x.hour <= 16:
+                space = 'часа'
+        if x.hour == 1 or x.hour == 13:
+                space = 'час'     
+        print(f'{dh[x.hour]} {space} ровно')
+    if 0 < x.minute < 30:
+        if firstdigit == 2:
+            print(f'{dm[20]} {dm[lastdigit]} {space} {d1[x.hour + 1]}')
+        else: print(f'{dm[x.minute]} {space} {d1[x.hour + 1]}')
+    if x.minute == 30:
+        print(f'половина {d1[x.hour + 1]}')
+    if 30 < x.minute < 45:
+        if firstdigit == 3:
+            print(f'{dm[30]} {dm[lastdigit]} {space} {d1[x.hour + 1]}')
+        else: print(f'{dm[40]} {dm[lastdigit]} {space} {d1[x.hour + 1]}')
+    if x.minute >= 45:
+        space = dm[60 - x.minute]
+        if x.hour == 0 or x.hour == 12:
+            dh[x.hour + 1] = 'час'
+        if 60 - x.minute == 1:
+            space = space.replace('а', 'ой')
+            print(f'без {space} минуты {dh[x.hour + 1]}')  
+        if 60 - x.minute == 2:
+            space = space.replace('е', 'ух')
+            print(f'без {space} минут {dh[x.hour + 1]}')
+        if 60 - x.minute == 3:
+            space = space.replace('и', 'ех')
+            print(f'без {space} минут {dh[x.hour + 1]}')
+        if 60 - x.minute == 4:
+            space = space + 'х'
+            print(f'без {space} минут {dh[x.hour + 1]}')
+        if 60 - x.minute == 8:
+            space = space.replace('емь', 'ьми')
+            print(f'без {space} минут {dh[x.hour + 1]}')
+        if 5 <= 60 - x.minute <= 7 or 9 <= 60 - x.minute <=15:
+            space = space.replace('ть', 'ти')
+            print(f'без {space} минут {dh[x.hour + 1]}')
```

Comment:
it's basically the same logic as above, it's a huge problem, there should be no copypast in a solution, you could've escape it by selecting a proper time value with if/else and describing the logic only once, -1 point (it's a huge mistake alright, but it can be easily fixed, so I will not take much)